src/js/node/net.ts

```diff
@@ -494,6 +494,16 @@ function Socket(options?) {
   if (options?.writableObjectMode)
     throw $ERR_INVALID_ARG_VALUE("options.writableObjectMode", options.writableObjectMode, "is not supported");
 
+  if (options?.fd !== undefined) {
+    if (options.fd < 0) {
+      throw $ERR_OUT_OF_RANGE("options.fd", options.fd, 0);
+    }
+
+    if (typeof options.fd !== "number") {
+      throw $ERR_INVALID_ARG_TYPE("options.fd", options.fd, "number");
+    }
+  }
```

Comment:
let's use the same validation as node
```suggestion
  if (options?.fd !== undefined) {
    validateInt32(options.fd, "options.fd", 0);
  }
```

Reply:
Fixed in 1880b66603f0747023a1b56e5cecac53fbb9bde4